### PR TITLE
Added #file, song[:file] and #genre

### PR DESCRIPTION
**Pull request by ap4y**
I have implemented `#file`, `#genre` and `song[:file]`. `#file` returns nil for multiple files cuesheet and `song[:file]` returns nil for single file cuesheet. `#genre` parses optionally remark field.

### CODE REVIEW

File: lib/rubycue/cuesheet.rb [hunk 1]
```diff
@@ -82,6 +86,21 @@ def parse_indices
...
 
+    def parse_files
+      unless @files
+        @files = cuesheet_scan(:file).map{|file| file.first}
+        @file = @files.delete_at(0) if @files.size == 1
+      end
+      @files
+    end
+
+    def parse_genre
```
Comment (blakesmith) [inline lib/rubycue/cuesheet.rb:97]: Is there a reason to iterate through all matches here, and then break? Can a cuesheet have more than one genre? Would it be easier to just write `@cuesheet.scan(@reg[:genre]).first` ?

File: lib/rubycue/cuesheet.rb [hunk 2]
```diff
@@ -19,7 +21,9 @@ def parse!
         song[:performer] = parse_performers[i]
         song[:track] = parse_tracks[i]
         song[:index] = parse_indices[i]
+        song[:file] = parse_files[i]
       end
+      parse_genre()
```
Comment (blakesmith) [inline lib/rubycue/cuesheet.rb]: Can you trim the parens here?

**Comment by blakesmith** [comment]
This looks great! Thanks for the additional features. Can you take a look at my comments and see what you think?

Cheers!

**Comment by ap4y** [comment]
Thanks for the comments! I have removed parentheses, sorry for that, old habits :(
About block iterator: since our regex contains groups I would have to do something like that:

``` ruby
scan_result = @cuesheet.scan(@reg[:genre])
if scan_result && scan_result.first
  first_array = scan_result.first
  @genre = first_array.first
end
```

So I thought that structure with block looks better. Also, as I understand sources of the `#scan` method, it should be faster, since we would break the loop on first encounter of the genre comment (mostly third row in cuesheet), without block we would iterate the whole cuesheet.

**Comment by blakesmith** [comment]
Awesome. Thanks for the contributions! Merged.

**Comment by ap4y** [comment]
Thank you for the project! It was really useful for me. Glad to help.

**Pull request merged**